src/lib/bootloader/grub2base.rb

```diff
@@ -120,6 +120,9 @@ def propose
 
       propose_serial
 
+      # we proposing new config and we do not know sections yet, so propose empty
+      @sections.default = ""
```

Comment:
To put it in yet another way: I suspect that we do not have a consistent state, either before this assignment, or after it.